kornia/augmentation/policy/__init__.py

```diff
@@ -0,0 +1 @@
+from .randaugment import RandAugment
```

Comment:
@shijianjian you mean to move this e.g to `dda` ? /cc @moskomule

Reply:
Then, dda is not differentiable data augmentation anymore 😉
Well, but splitting specific augmentation methods to others is good to keep kornia simple, I think.

Reply:
@edgarriba I think it is closer to dda functions as it used ```magnitude``` as the parameter. Similar logic to ```dda``` at the current stage.

@moskomule The kornia augmentation APIs are also differentiable (apart from those uint8 based operations). I think we may have another name than ```dda``` like ```meta_augment``` or something? I think it can be:
```python
from kornia import meta_augment
meta_augment.nn (dda.operations)
meta_augment.nn.functional (dda.functional)
meta_augment.policy.RandAugment
meta_augment.policy.FasterAutoAugment.search()
meta_augment.policy.FasterAutoAugment(res.yml)
```

Reply:
Or `from kornia.augment import search_strategy`, which I think is easier to understand. IIRC, RandAugment also needs search, so we can create more abstract thing like:

```python
from kornia.augment import search_strategy

search_strategy.RandAugment(depth=2).search(export_as='ra.yaml')
search_strategy.FasterAutoAugment(depth=2).search(export_as='faa.yaml')
```

Reply:
Indeed, I think ```strategy``` is a better word here. RandAugment/AutoAugment need to search but in a non-differentiable way, also very slow as far as what I know. I am not sure if it is worth implementing. I think at least, I want to include the search function for FasterAutoAugment, and it seems pretty quick (2 or 3 hours) by using your dda repo.


For the naming, I prefer removing the ```search``` part since the search part will be implemented inside each class.
```python
from kornia.augmentation import strategy

faas = strategy.FasterAutoAugmentStrategy(methods=[solarize, etc.], depth=[2, 4, 6])  # search space is specified here
policy = faas.search(temperature=0.1)   # some hparams for searching, like temperature
 # load and continue searching with other hparams
faas2 = FasterAutoAugmentStrategy.load(policy) 
policy2 = faas2.search(temperature=0.5)

strategy.save_strategy(policy2, name='faa.yml')
# load policy for network usage
faa = strategy.FasterAutoAugment(policy='faa.yml')
# or
faa = strategy.FasterAutoAugment(policy=policy2)
```

For the abstraction, I am thinking of
```python
class FasterAutoAugment(AugmentationBase2D):
    def __init__(self, policy):
        if policy.endswith('.yml'):
            self.policy = read_yml(policy)
        else:
            self.policy = policy
    ...
    def forward(self, input):
         ...

class AugmentationStrategy(nn.Module):
    def __init__(self, policy):
        if policy.endswith('.yml'):
            self.policy = read_yml(policy)
        else:
            self.policy = policy
    @classmethod
    def load(cls, policy):
        return cls(policy)
    def search(self, methods, depth, temperature):
        raise NotImplementedError
    def construct(self,):
        # Something like
        # return Augmentation(self.policy)
        raise NotImplementedError
```

@moskomule you are more expert in autoaugment than me. What do you think?

Reply:
this is super cool. Does it mean we are creating here an abstract API that can consume any augmentation ? I mean that won't anything specific to faster paper or any other ?

Reply:
`faas = strategy.FasterAutoAugmentStrategy(methods=[solarize, etc.], depth=[2, 4, 6])  # search space is specified here` is great and allows users to design more flexible augmentation strategies based on their data, resource, etc.
I think it's useful to have `AugmentationStrategy.pretrained(dataset_name)`, especially for AutoAugment, or such expensive ones (cf. https://github.com/tensorflow/tpu/blob/4489103eee24cd612ea3c41958c9320c45080c6f/models/official/efficientnet/autoaugment.py#L36)

Reply:
Great. I will modify this PR to align our discussion. Then we may add the FasterAutoAugment on top on that. Will let you both know when I finish it.